packages/object/src/Charcoal/Object/GenerateRevisionListener.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace Charcoal\Object;
+
+use Charcoal\Event\AbstractEventListener;
+use Charcoal\Model\ModelInterface;
+use Pimple\Container;
+
+/**
+ * Listener
+ */
+class GenerateRevisionListener extends AbstractEventListener
+{
+    protected RevisionService $revisionService;
+
+    public function __invoke(object $event)
```

Comment:
I think this should be type-hinted to `Charcoal\Event\Events\Object\AbstractObjectEvent` which, at this point, should also be an interface (`ObjectEventInterface` that declares `getObject()`).

```suggestion
    public function __invoke(object $event)
```

Reply:
I like the interface thing. I'll refactor object events in the event branch